### ISSUE TITLE
lib-user: stats select menus don't have accessible names

### ISSUE THREAD

**Issue by eatyourgreens** [zooniverse/front-end-monorepo#6603]
## Package
- lib-user

## Describe the bug
The user stats dropdown menus don't have particularly accessible names:
<img width="900" alt="The Firefox accessibility tree for user stats. The project dropdown is called project-select; Selected [object Object]. The ddate range dropdown is called data-range-select; Selected: [object Object]." src="https://github.com/user-attachments/assets/c4ec2a5a-b9e5-4064-94b0-8704aab8fdd7" />

## To Reproduce
Go to your stats and inspect the accessibility properties of each dropdown menu in the browser dev tools. Alternatively, start VoiceOver (Narrator on Windows) and navigate to each menu in turn.

## Expected behavior
The menus should have clear, understandable labels.

## Additional context
[WCAG 2.2: Success Criterion 3.3.2: clear labels and instructions](https://www.w3.org/WAI/WCAG22/Understanding/labels-or-instructions.html) is a Level A requirement.

**Comment by eatyourgreens** [comment]
The group visibility select menu also has an inaccessible name: 'Stats Visibility; Selected: private_show_agg_and_ind'.

<img width="900" alt="Accessibility tree for the group visibility menu, showing that its name is Stats Visibility; Selected: private_show_agg_and_ind" src="https://github.com/user-attachments/assets/44901bfb-5ff5-4e09-bc5d-37a45fa8ac46" />
